feat(modelarmor): Added samples for Model Armor service

harshnasitcrest opened the pull request:
Added following samples for Model Armor:

1. Quickstart & Add CODEOWNERS
2. Create a template
3. Create a template with labels
4. Create a template with its metadata
5. Create a template with Basic SDP configuration
6. Create a template with Advanced SDP configuration
7. Update template configuration for given template_id
8. Update template metadata- Update labels of a template
9. View Template
10. List all Templates
11. Delete Template
12. Sanitize the user prompt based on the given template configuration
13. Scan a PDF file based on the given template configuration
14. Sanitize the model response based on the given template configuration
15. Sanitize the model response along with user prompt
16. Get Project Floor Settings
17. Get Folder Floor Settings
18. Get Organization Floor Settings
19. Update Project Floor Settings
20. Update Folder Floor Settings
21. Update Organization Floor Settings

Please enable the Model Armor API and grant `modelarmor.admin` IAM permission on the service account/project that is used in this repo.

bshaffer reviewed (changes requested):
These samples are the old style - the new style (except in `quickstart.php` is to wrap in a function. See https://github.com/GoogleCloudPlatform/php-docs-samples/pull/2105 for example.

Also, your tests could be simplified tremendously. I'll try to find time this week to refactor them.

bshaffer commented on modelarmor/test/BaseTestCase.php:
```diff
@@ -0,0 +1,73 @@
+        return $output;
+    }
+
+    protected static function getProjectId()
+    {
+        return self::$projectId;
+    }
```
This also isn't necessary, why not just reference `self::$projectId`?
```suggestion
    protected static function getProjectId()
    {
        return self::$projectId;
    }

```

bshaffer commented on modelarmor/test/quickstartTest.php:
```diff
@@ -0,0 +1,43 @@
+    public function testQuickstart()
+    {
+        $projectId = self::getProjectId();
+        $output = $this->runSnippetfile('quickstart', [
```
```suggestion
        $output = $this->runSnippet('quickstart', [

```

harshnasitcrest replied:
Done

bshaffer commented on modelarmor/test/BaseTestCase.php:
```diff
@@ -0,0 +1,73 @@
+                throw $e;
+            }
+        }
+        self::$client->close();
```
I do think this would be much easier in a single test file... e.g.  `tests/modelArmorTemplateTests.php`. But if you prefer it this way, that's also fine by me...

harshnasitcrest replied:
Done.

bshaffer commented:
> As per a repo owner’s suggestion, we divided the PRs in chunks to ease review. However, we spent extra effort resolving merge conflicts on the tests file every time a PR was merged to main. So, we decided to divide the tests per snippet, along with common code or fixture, in one file. I’ve added the conversation for context.

I personally would prefer one big PR, but again, this is not a blocker, just an FYI.

harshnasitcrest commented:
> > As per a repo owner’s suggestion, we divided the PRs in chunks to ease review. However, we spent extra effort resolving merge conflicts on the tests file every time a PR was merged to main. So, we decided to divide the tests per snippet, along with common code or fixture, in one file. I’ve added the conversation for context.
> 
> I personally would prefer one big PR, but again, this is not a blocker, just an FYI.

Thank you for the clarification. If you prefer, I will close all other PRs and create a single, comprehensive PR. This approach would facilitate review and minimise the time required to resolve merge conflicts.

bshaffer commented:
@harshnasitcrest that would be really great! As long as it doesn't cause too much additional work for you.

Thanks!

harshnasitcrest commented:
> @harshnasitcrest that would be really great! As long as it doesn't cause too much additional work for you.
> 
> Thanks!

I've addressed all the comments, merged all the samples in this PR along with tests in one file. Hence deprecating and closing all other open PRs of Model Armor.

bshaffer commented on modelarmor/src/create_template.php:
```diff
@@ -0,0 +1,84 @@
+    $client = new ModelArmorClient($options);
+    $parent = $client->locationName($projectId, $locationId);
+
+    /** Build the Model Armor template with preferred filters.
```
```suggestion
    /**
     * Build the Model Armor template with preferred filters.
```

bshaffer commented on modelarmor/src/create_template_with_advanced_sdp.php:
```diff
@@ -0,0 +1,75 @@
+ * @param string $inspectTemplate The resource name of the inspect template. (e.g. 'organizations/{organization}/inspectTemplates/{inspect_template}')
+ * @param string $deidentifyTemplate The resource name of the de-identify template. (e.g. 'organizations/{organization}/deidentifyTemplates/{deidentify_template}')
+ */
+function create_template_with_advanced_sdp(string $projectId, string $locationId, string $templateId, string $inspectTemplate, string $deidentifyTemplate):void
```
```suggestion
function create_template_with_advanced_sdp(
    string $projectId,
    string $locationId,
    string $templateId,
    string $inspectTemplate,
    string $deidentifyTemplate
): void
```

bshaffer commented on modelarmor/src/create_template_with_advanced_sdp.php:
```diff
@@ -0,0 +1,75 @@
+ * @param string $projectId The ID of the project (e.g. 'my-project').
+ * @param string $locationId The ID of the location (e.g. 'us-central1').
+ * @param string $templateId The ID of the template (e.g. 'my-template').
+ * @param string $inspectTemplate The resource name of the inspect template. (e.g. 'organizations/{organization}/inspectTemplates/{inspect_template}')
+ * @param string $deidentifyTemplate The resource name of the de-identify template. (e.g. 'organizations/{organization}/deidentifyTemplates/{deidentify_template}')
```
```suggestion
 * @param string $inspectTemplate The resource name of the inspect template.
          (e.g. 'organizations/{organization}/inspectTemplates/{inspect_template}')
 * @param string $deidentifyTemplate The resource name of the de-identify template. 
          (e.g. 'organizations/{organization}/deidentifyTemplates/{deidentify_template}')
```

bshaffer commented on modelarmor/src/create_template_with_labels.php:
```diff
@@ -0,0 +1,83 @@
+ * @param string $labelKey The key of the label to add. (e.g. 'my-label-key').
+ * @param string $labelValue The value of the label to add. (e.g. 'my-label-value').
+ */
+function create_template_with_labels(string $projectId, string $locationId, string $templateId, string $labelKey, string $labelValue):void
```
```suggestion
function create_template_with_labels(
    string $projectId,
    string $locationId,
    string $templateId,
    string $labelKey,
    string $labelValue
): void
```

bshaffer commented on modelarmor/src/sanitize_model_response.php:
```diff
@@ -0,0 +1,52 @@
+ * @param string $templateId The ID of the template (e.g. 'my-template').
+ * @param string $modelResponse The model response to sanitize (e.g. 'my-model-response').
+ */
+function sanitize_model_response($projectId, $locationId, $templateId, $modelResponse):void
```
```suggestion
function sanitize_model_response(
    string $projectId,
    string $locationId,
    string $templateId,
    string $modelRespons
): void
```

bshaffer commented on modelarmor/src/sanitize_model_response_with_user_prompt.php:
```diff
@@ -0,0 +1,54 @@
+ * @param string $modelResponse The model response to sanitize (e.g. 'my-model-response').
+ * @param string $userPrompt The user prompt for the model response (e.g. 'my-user-prompt').
+ */
+function sanitize_model_response_with_user_prompt($projectId, $locationId, $templateId, $modelResponse, $userPrompt):void
```
```suggestion
function sanitize_model_response_with_user_prompt(
    string $projectId,
    string $locationId,
    string $templateId,
    string $modelResponse,
    string $userPrompt
): void
```

bshaffer commented on modelarmor/src/sanitize_user_prompt.php:
```diff
@@ -0,0 +1,52 @@
+ * @param string $templateId The ID of the template (e.g. 'my-template').
+ * @param string $userPrompt The user prompt to sanitize (e.g. 'my-user-prompt').
+ */
+function sanitize_user_prompt($projectId, $locationId, $templateId, $userPrompt):void
```
```suggestion
function sanitize_user_prompt(
    string $projectId,
    string $locationId,
    string $templateId,
    string $userPrompt
): void
```

bshaffer commented on modelarmor/src/screen_pdf_file.php:
```diff
@@ -0,0 +1,67 @@
+ * @param string $templateId The ID of the template to use for the screener (e.g. 'my-template').
+ * @param string $filePath The path to the PDF file to screen (e.g. 'path/to/file.pdf').
+ */
+function screen_pdf_file($projectId, $locationId, $templateId, $filePath):void
```
```suggestion
function screen_pdf_file(
    string $projectId,
    string $locationId,
    string $templateId,
    string $filePath
): void
```

bshaffer commented on modelarmor/src/update_folder_floor_settings.php:
```diff
@@ -0,0 +1,78 @@
+ * @param string $folderId The folder Id for which the floor settings is to be updated.
+ *
+ */
+function update_folder_floor_settings($folderId) {
```
```suggestion
function update_folder_floor_settings(string $folderId): void
{
```

bshaffer commented on modelarmor/src/update_organization_floor_settings.php:
```diff
@@ -0,0 +1,78 @@
+ * @param string $organizationId The organization Id for which the floor settings is to be updated.
+ *
+ */
+function update_organization_floor_settings($organizationId) {
```
```suggestion
function update_organization_floor_settings(string $organizationId)
{
```

bshaffer commented on modelarmor/src/update_template_labels.php:
```diff
@@ -0,0 +1,62 @@
+ * @param string $labelKey The key of the label to add (e.g. 'my-label-key').
+ * @param string $labelValue The value of the label to add (e.g. 'my-label-value').
+ */
+function update_template_labels($projectId, $locationId, $templateId, $labelKey, $labelValue) {
```
```suggestion
function update_template_labels(
    string $projectId,
    string $locationId,
    string $templateId,
    string $labelKey,
    string $labelValue
): void {
```

bshaffer commented on modelarmor/src/update_template_metadata.php:
```diff
@@ -0,0 +1,75 @@
+ * @param string $locationId The ID of the location (e.g. 'us-central1').
+ * @param string $templateId The ID of the template (e.g. 'my-template').
+ */
+function update_template_metadata($projectId, $locationId, $templateId): void
```
```suggestion
function update_template_metadata(string $projectId, string $locationId, string $templateId): void
```

bshaffer commented on modelarmor/src/delete_template.php:
```diff
@@ -0,0 +1,49 @@
+ * @param string $locationId The ID of the location (e.g. 'us-central1').
+ * @param string $templateId The ID of the template (e.g. 'my-template').
+ */
+function delete_template($projectId, $locationId, $templateId) :void
```
```suggestion
function delete_template(string $projectId, string $locationId, string $templateId): void
```

bshaffer commented on modelarmor/src/update_folder_floor_settings.php:
```diff
@@ -0,0 +1,78 @@
+
+namespace Google\Cloud\Samples\ModelArmor;
+
+require_once __DIR__ . '/../vendor/autoload.php';
+
+if (count($argv) != 2) {
+    return printf("Usage: php %s FOLDER_ID\n", basename(__FILE__));
+}
+list($_, $folderId) = $argv;
+
+
```
```suggestion
```

bshaffer commented on modelarmor/src/update_organization_floor_settings.php:
```diff
@@ -0,0 +1,78 @@
+
+namespace Google\Cloud\Samples\ModelArmor;
+
+require_once __DIR__ . '/../vendor/autoload.php';
+
+if (count($argv) != 2) {
+    return printf("Usage: php %s ORGANIZATION_ID\n", basename(__FILE__));
+}
+list($_, $organizationId) = $argv;
+
+
```
```suggestion
```

bshaffer commented on modelarmor/src/get_folder_floor_settings.php:
```diff
@@ -0,0 +1,46 @@
+ * @param string $folderId The folder Id for which the floor settings is to be retrieved.
+ *
+ */
+function get_folder_floor_settings($folderId) 
```
```suggestion
function get_folder_floor_settings(string $folderId): void
```

bshaffer commented on modelarmor/src/get_organization_floor_settings.php:
```diff
@@ -0,0 +1,46 @@
+ * @param string $organizationId The organization Id for which the floor settings is to be retrieved.
+ *
+ */
+function get_organization_floor_settings($organizationId)
```
```suggestion
function get_organization_floor_settings(string $organizationId): void
```

bshaffer commented on modelarmor/src/get_project_floor_settings.php:
```diff
@@ -0,0 +1,46 @@
+ * @param string $projectId The project Id for which the floor settings is to be retrieved.
+ *
+ */
+function get_project_floor_settings($projectId)
```
```suggestion
function get_project_floor_settings(string $projectId): void
```

bshaffer commented on modelarmor/src/get_template.php:
```diff
@@ -0,0 +1,54 @@
+ * @param string $locationId The ID of the location where the template is stored (e.g. 'us-central1').
+ * @param string $templateId The ID of the template (e.g. 'my-template').
+ */
+function get_template($projectId, $locationId, $templateId): void
```
```suggestion
function get_template(string $projectId, string $locationId, string $templateId): void
```

bshaffer commented on modelarmor/src/list_templates.php:
```diff
@@ -0,0 +1,51 @@
+ * @param string $projectId The ID of your Google Cloud Platform project (e.g. 'my-project').
+ * @param string $locationId The ID of the location where the templates are stored (e.g. 'us-central1').
+ */
+function list_templates($projectId, $locationId): void
```
```suggestion
function list_templates(string $projectId, string $locationId): void
```

bshaffer reviewed (approved):
minor style changes and removal of unneeded helper code
